test/Microsoft.ML.Benchmarks/Helpers/DataPathProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.ML.Benchmarks
+{
+    internal class DataPathProvider
+    {
+        private static string RootPath = Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.Parent.Parent.Parent.Parent.Parent.FullName;
```

Comment:
These values are duplicated with [https://github.com/dotnet/machinelearning/blob/64b60d2805ba4ef1fd694c3211d170fdf0699c3b/test/Microsoft.ML.TestFramework/BaseTestClass.cs#L33-L44](https://github.com/dotnet/machinelearning/blob/64b60d2805ba4ef1fd694c3211d170fdf0699c3b/test/Microsoft.ML.TestFramework/BaseTestClass.cs#L33-L44).

Can we refactor this code so it is contained in 1 spot?

Reply:
This is not possible as the currentAssembly locations are different for benchmark tests and other tests,
eg for all other tests, the currentAssemblyLocation = 
```"C:\\git\\machinelearning\\bin\\AnyCPU.Release\\Microsoft.ML.Benchmarks\\netcoreapp2.1\\Microsoft.ML.TestFramework.dll"```
for benchmarks it is equal to 
```"C:\\git\\machinelearning\\bin\\AnyCPU.Release\\Microsoft.ML.Benchmarks\\netcoreapp2.1\\0c2f3461-1faf-408d-9864-57e4a29ec2f4\\bin\\Release\\netcoreapp2.1\\Microsoft.ML.TestFramework.dll"```

Reply:
Maybe a better approach here would be to navigate up until we hit the "repo root" directory. Then navigate down to `test\data`. Instead of hard-coding how many parent directories until the root.

See 
https://github.com/dotnet/cli/blob/6da8a92712d114af8131c2683c0c0f78436f6d80/test/Microsoft.DotNet.Tools.Tests.Utilities/RepoDirectoriesProvider.cs#L24-L57
or
https://github.com/dotnet/sdk/blob/67275f2a66b71e480fcc36b17a5b4b0e06f8e5da/src/Tests/Microsoft.NET.TestFramework/TestContext.cs#L156-L170 

for examples of how to do this.